vtm/src/org/oscim/layers/vector/AbstractVectorLayer.java

```diff
@@ -21,8 +40,11 @@
 
     protected final static double UNSCALE_COORD = 4;
 
+    // limit coords to maximum resolution of GL.Short
+    private final int MAX_CLIP = (int) ((Tile.SIZE * Tile.TILE_SIZE_MULTIPLE) / MapRenderer.COORD_SCALE) - 1;
```

Comment:
Can you explain in detail the principle for this calculation?

Also why is need to multiply `Tile.TILE_SIZE_MULTIPLE`, since its only purpose is in `calculateTileSize` as preset, to clamp tile size to a value multiple.

Reply:
I'm really sorry, I messed up something...

In [LineBucket](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/renderer/bucket/LineBucket.java#L213) the values are multiplied with `COORD_SCALE`. So the clipper should clip exactly that value, so that `value * COORD_SCALE <= Short.MAX_VALUE = 32767` 

Example:
```
float value = 5000f;
// 4095 = ((Short.MAX_VALUE) / 8)    // for COORD_SCALE = 8
// or 8192 = ((Short.MAX_VALUE) / 4)  // for COORD_SCALE = 4
int MAX_CLIP = (int) ((Tile.SIZE * Tile.TILE_SIZE_MULTIPLE) / MapRenderer.COORD_SCALE) - 1;

clipper.clip(value);
// 32767 = 4095 * 8 // for COORD_SCALE = 8
// 32767 = 8192 * 4 // for COORD_SCALE = 4
out = value * COORD_SCALE;
```
I'll change that.